src/datachain/query/dataset.py

```diff
@@ -1175,8 +1175,12 @@ def attached(self) -> bool:
         """
         return self.name is not None and self.version is not None
 
-    def c(self, name: Union[C, str]) -> "ColumnClause[Any]":
-        col = sqlalchemy.column(name) if isinstance(name, str) else name
+    def c(self, column: Union[C, str]) -> "ColumnClause[Any]":
+        col: sqlalchemy.ColumnClause = (
+            sqlalchemy.column(column)
+            if isinstance(column, str)
+            else sqlalchemy.column(column.name, column.type)
```

Comment:
[F] It does not seem possible to overwrite the table of a column which already has the table set. I.e `test_merge_with_itself_column` fails without this change.

Reply:
This change is no longer required for this PR to work. I'd be happy to split it out if required.

Reply:
As for me personally I am OK to leave it here. It feels natural to have this change here.